[lint] Consider using f string

With python 3.10 and the latest pylint 2.11, the recommendation for f string is now default:

https://pylint.pycqa.org/en/latest/whatsnew/2.11.html#new-checkers

```
(venv) x1 ➜  ansible-navigator git:(lint_errors) pylint src/ansible_navigator --ignore=tm_tokenize --disable=all --enable=consider-using-f-string
************* Module ansible_navigator.utils
src/ansible_navigator/utils.py:355:15: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
src/ansible_navigator/utils.py:357:15: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
src/ansible_navigator/utils.py:359:15: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
src/ansible_navigator/utils.py:360:11: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
************* Module ansible_navigator.steps
src/ansible_navigator/steps.py:115:29: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
************* Module ansible_navigator.configuration_subsystem.navigator_configuration
src/ansible_navigator/configuration_subsystem/navigator_configuration.py:60:18: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
************* Module ansible_navigator.actions.inventory
src/ansible_navigator/actions/inventory.py:62:13: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
************* Module ansible_navigator.actions.doc
src/ansible_navigator/actions/doc.py:185:22: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
************* Module ansible_navigator.actions.run
src/ansible_navigator/actions/run.py:112:17: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
src/ansible_navigator/actions/run.py:118:17: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
src/ansible_navigator/actions/run.py:136:17: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
************* Module ansible_navigator.ui_framework.ui
src/ansible_navigator/ui_framework/ui.py:248:21: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
src/ansible_navigator/ui_framework/ui.py:261:20: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
src/ansible_navigator/ui_framework/ui.py:406:25: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
```

--- a/src/ansible_navigator/actions/doc.py
+++ b/src/ansible_navigator/actions/doc.py
@@ -182,8 +182,9 @@ class Action(App):
                 [self._plugin_name], plugin_type=self._plugin_type, playbook_dir=playbook_dir
             )
             if plugin_doc_err:
-                msg = "Error occurred while fetching doc for" " plugin {0}: '{1}'".format(
-                    self._plugin_name, plugin_doc_err
+                msg = (
+                    "Error occurred while fetching doc for plugin"
+                    f" {self._plugin_name}: '{plugin_doc_err}'"
                 )
                 self._logger.error(msg)
 

--- a/src/ansible_navigator/actions/doc.py
+++ b/src/ansible_navigator/actions/doc.py
@@ -182,12 +182,11 @@ class Action(App):
                 [self._plugin_name], plugin_type=self._plugin_type, playbook_dir=playbook_dir
             )
             if plugin_doc_err:
-                msg = (
-                    "Error occurred while fetching doc for plugin"
-                    f" {self._plugin_name}: '{plugin_doc_err}'"
+                self._logger.error(
+                    "Error occurred while fetching doc for plugin %s: '%s'",
+                    self._plugin_name,
+                    plugin_doc_err,
                 )
-                self._logger.error(msg)
-
             plugin_doc_response = self._extract_plugin_doc(plugin_doc, plugin_doc_err)
             return plugin_doc_response
         else:

--- a/src/ansible_navigator/actions/inventory.py
+++ b/src/ansible_navigator/actions/inventory.py
@@ -59,10 +59,9 @@ def content_heading(obj: Any, screen_w: int) -> Union[CursesLines, None]:
     """
 
     heading = []
-    string = "[{host}] {os}".format(
-        host=obj["inventory_hostname"],
-        os=obj.get("ansible_network_os", obj.get("ansible_platform", "")),
-    )
+    host = obj["inventory_hostname"]
+    operating_system = obj.get("ansible_network_os", obj.get("ansible_platform", ""))
+    string = f"[{host}] {operating_system}"
     string = string + (" " * (screen_w - len(string) + 1))
     heading.append(
         tuple(

--- a/src/ansible_navigator/actions/run.py
+++ b/src/ansible_navigator/actions/run.py
@@ -109,13 +109,13 @@ def content_heading(obj: Any, screen_w: int) -> Union[CursesLines, None]:
 
     if isinstance(obj, dict) and "task" in obj:
         heading = []
-        detail = "PLAY [{play}:{tnum}] ".format(play=obj["play"], tnum=obj["__number"])
+        detail = f"PLAY [{obj['play']}:{obj['__number']}] "
         stars = "*" * (screen_w - len(detail))
         heading.append(
             tuple([CursesLinePart(column=0, string=detail + stars, color=0, decoration=0)])
         )
 
-        detail = "TASK [{task}] ".format(task=obj["task"])
+        detail = f"TASK [{obj['task']}] "
         stars = "*" * (screen_w - len(detail))
         heading.append(
             tuple([CursesLinePart(column=0, string=detail + stars, color=0, decoration=0)])
@@ -133,7 +133,7 @@ def content_heading(obj: Any, screen_w: int) -> Union[CursesLines, None]:
         else:
             msg = ""
 
-        string = "{res}: [{host}] {msg}".format(res=res, host=obj["__host"], msg=msg)
+        string = f"{res}: [{obj['__host']}] {msg}"
         string = string + (" " * (screen_w - len(string) + 1))
         heading.append(
             tuple(

--- a/src/ansible_navigator/configuration_subsystem/navigator_configuration.py
+++ b/src/ansible_navigator/configuration_subsystem/navigator_configuration.py
@@ -57,7 +57,7 @@ def generate_editor_command() -> str:
     else:
         message = "EDITOR environment variable set as '{editor}'"
         initialization_messages.append(LogMessage(level=logging.DEBUG, message=message))
-        command = "%s {filename}" % os.environ.get("EDITOR")
+        command = f"{os.environ.get('EDITOR')} {{filename}}"
     message = f"Default editor_command set to: {command}"
     initialization_messages.append(LogMessage(level=logging.DEBUG, message=message))
     return command

--- a/src/ansible_navigator/configuration_subsystem/navigator_configuration.py
+++ b/src/ansible_navigator/configuration_subsystem/navigator_configuration.py
@@ -55,9 +55,9 @@ def generate_editor_command() -> str:
         initialization_messages.append(LogMessage(level=logging.DEBUG, message=message))
         command = "vi +{line_number} {filename}"
     else:
-        message = "EDITOR environment variable set as '{editor}'"
+        message = f"EDITOR environment variable set as '{editor}'"
         initialization_messages.append(LogMessage(level=logging.DEBUG, message=message))
-        command = f"{editor} {{filename}}"
+        command = f"{editor!s} {{filename}}"
     message = f"Default editor_command set to: {command}"
     initialization_messages.append(LogMessage(level=logging.DEBUG, message=message))
     return command

--- a/src/ansible_navigator/configuration_subsystem/navigator_configuration.py
+++ b/src/ansible_navigator/configuration_subsystem/navigator_configuration.py
@@ -57,7 +57,7 @@ def generate_editor_command() -> str:
     else:
         message = "EDITOR environment variable set as '{editor}'"
         initialization_messages.append(LogMessage(level=logging.DEBUG, message=message))
-        command = f"{os.environ.get('EDITOR')} {{filename}}"
+        command = f"{editor} {{filename}}"
     message = f"Default editor_command set to: {command}"
     initialization_messages.append(LogMessage(level=logging.DEBUG, message=message))
     return command

--- a/src/ansible_navigator/steps.py
+++ b/src/ansible_navigator/steps.py
@@ -112,7 +112,7 @@ class Step:
     def _value_check(value: Any, want: Union[type, Tuple[type, ...]]) -> None:
         """check some expect type against a value"""
         if not isinstance(value, want):
-            raise ValueError("wanted {want}, got {value}".format(want=want, value=type(value)))
+            raise ValueError(f"wanted {want}, got {type(value)}")
 
 
 class Steps(deque):

--- a/src/ansible_navigator/ui_framework/ui.py
+++ b/src/ansible_navigator/ui_framework/ui.py
@@ -245,7 +245,7 @@ class UserInterface(CursesWindow):
         :return: The footer line
         :rtype: CursesLine
         """
-        colws = [len("{k}: {v}".format(k=str(k), v=str(v))) for k, v in key_dict.items()]
+        colws = [len(f"{str(k)}: {str(v)}") for k, v in key_dict.items()]
         if self._status:
             status_width = self._pbar_width
         else:
@@ -258,7 +258,7 @@ class UserInterface(CursesWindow):
         footer = []
         for idx, key in enumerate(key_dict):
             left = key[0 : adj_colws[idx]]
-            right = " {v}".format(v=key_dict[key])
+            right = f" {key_dict[key]}"
             right = right[0 : adj_colws[idx] - len(key)]
             footer.append(
                 CursesLinePart(
@@ -403,7 +403,8 @@ class UserInterface(CursesWindow):
             # Add the content
             for idx, line in enumerate(lines):
                 line_index = line_numbers[idx]
-                prefix = "{idx}\u2502".format(idx=str(line_index).rjust(index_width))
+                line_index_str = str(line_index).rjust(index_width)
+                prefix = f"{line_index_str}\u2502"
                 self._add_line(
                     window=self._screen, lineno=idx + len(heading), line=line, prefix=prefix
                 )

--- a/src/ansible_navigator/utils.py
+++ b/src/ansible_navigator/utils.py
@@ -352,7 +352,7 @@ def human_time(seconds: Union[int, float]) -> str:
     hours, seconds = divmod(seconds, 3600)
     minutes, seconds = divmod(seconds, 60)
     if days > 0:
-        return f"{sign_string}{days}d{hours}h{minutes}m{seconds}s"
+        return f"{sign_string!s}{days:d}d{hours:d}h{minutes}m{seconds}s"
     if hours > 0:
         return f"{sign_string}{hours}h{minutes}m{seconds}s"
     if minutes > 0:

--- a/src/ansible_navigator/utils.py
+++ b/src/ansible_navigator/utils.py
@@ -352,7 +352,7 @@ def human_time(seconds: Union[int, float]) -> str:
     hours, seconds = divmod(seconds, 3600)
     minutes, seconds = divmod(seconds, 60)
     if days > 0:
-        return f"{sign_string!s}{days:d}d{hours:d}h{minutes}m{seconds}s"
+        return f"{sign_string}{days}d{hours}h{minutes}m{seconds}s"
     if hours > 0:
         return f"{sign_string}{hours}h{minutes}m{seconds}s"
     if minutes > 0:

--- a/src/ansible_navigator/utils.py
+++ b/src/ansible_navigator/utils.py
@@ -352,12 +352,12 @@ def human_time(seconds: Union[int, float]) -> str:
     hours, seconds = divmod(seconds, 3600)
     minutes, seconds = divmod(seconds, 60)
     if days > 0:
-        return "%s%dd%dh%dm%ds" % (sign_string, days, hours, minutes, seconds)
+        return f"{sign_string}{days}d{hours}h{minutes}m{seconds}s"
     if hours > 0:
-        return "%s%dh%dm%ds" % (sign_string, hours, minutes, seconds)
+        return f"{sign_string}{hours}h{minutes}m{seconds}s"
     if minutes > 0:
-        return "%s%dm%ds" % (sign_string, minutes, seconds)
-    return "%s%ds" % (sign_string, seconds)
+        return f"{sign_string}{minutes}m{seconds}s"
+    return f"{sign_string}{seconds}s"
 
 
 PASCAL_REGEX = re.compile("((?<=[a-z0-9])[A-Z]|(?!^)[A-Z](?=[a-z]))")

--- a/src/ansible_navigator/utils.py
+++ b/src/ansible_navigator/utils.py
@@ -352,12 +352,12 @@ def human_time(seconds: Union[int, float]) -> str:
     hours, seconds = divmod(seconds, 3600)
     minutes, seconds = divmod(seconds, 60)
     if days > 0:
-        return f"{sign_string}{days}d{hours}h{minutes}m{seconds}s"
+        return f"{sign_string!s}{days:d}d{hours:d}h{minutes:d}m{seconds:d}s"
     if hours > 0:
-        return f"{sign_string}{hours}h{minutes}m{seconds}s"
+        return f"{sign_string!s}{hours:d}h{minutes:d}m{seconds:d}s"
     if minutes > 0:
-        return f"{sign_string}{minutes}m{seconds}s"
-    return f"{sign_string}{seconds}s"
+        return f"{sign_string!s}{minutes:d}m{seconds:d}s"
+    return f"{sign_string!s}{seconds:d}s"
 
 
 PASCAL_REGEX = re.compile("((?<=[a-z0-9])[A-Z]|(?!^)[A-Z](?=[a-z]))")

--- a/src/ansible_navigator/utils.py
+++ b/src/ansible_navigator/utils.py
@@ -342,15 +342,18 @@ def get_share_directory(app_name) -> Tuple[List[LogMessage], List[ExitMessage], 
     exit_messages.append(ExitMessage(message=exit_msg))
     return messages, exit_messages, None
 
+def divmod_int(numerator: Union[int, float], denominator: Union[int, float]) -> Tuple[int, int]:
+    """Return the result of divmod, as a tuple of integers."""
+    quotient, remainder = divmod(numerator, denominator)
+    return int(quotient), int(remainder)
 
-def human_time(seconds: Union[int, float]) -> str:
-    """convert seconds into human readable
-    00d00h00m00s format"""
+def human_time(seconds: int) -> str:
+    """Convert seconds into human readable 00d00h00m00s format."""
     sign_string = "-" if seconds < 0 else ""
     seconds = abs(int(seconds))
-    days, seconds = divmod(seconds, 86400)
-    hours, seconds = divmod(seconds, 3600)
-    minutes, seconds = divmod(seconds, 60)
+    days, seconds = divmod_int(seconds, 86400)
+    hours, seconds = divmod_int(seconds, 3600)
+    minutes, seconds = divmod_int(seconds, 60)
     if days > 0:
         return f"{sign_string!s}{days:d}d{hours:d}h{minutes:d}m{seconds:d}s"
     if hours > 0:

